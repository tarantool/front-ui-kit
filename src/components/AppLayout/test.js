import React from 'react';
import { css } from '@emotion/css';
import renderer from 'react-test-renderer';
import {
  AppHeader,
  AppLayout,
  Button,
  IconCluster,
  IconCode,
  IconTrash,
  IconInfo,
  IconSearch,
  SideMenu,
  SVGImage,
  TarantoolLogoCompact,
  TarantoolLogoFull
} from '../../index';

jest.mock(
  '../Scrollbar/index.js',
  () => ({
    Scrollbar: ({ children, className }) => <div className={className}>{children}</div>
  })
);

it('AppLayout renders correctly', () => {
  const onLinkClick = jest.fn();
  const onMenuItemClick = jest.fn();
  const toggleExpand = jest.fn();

  const logoStyle = css`margin-left: 24px;`;

  const breadcrumbs = [
    {
      title: '@tarantool.io/ui-kit',
      path: '/tarantool'
    },
    {
      title: 'UI Components',
      path: '/section-ui-components'
    },
    {
      title: 'Breadcrumb',
      path: '/breadcrumb'
    },
    {
      title: 'OverflowList',
      path: '/overflow-list'
    },
    {
      title: 'ResizeSensor',
      path: '/resize-sensor'
    }
  ];

  const menuItems = [
    {
      'selected': false,
      'expanded': false,
      'loading': false,
      'items': [],
      'icon': IconInfo,
      'label': 'My Test',
      'path': '/mytest/test',
      'namespace': 'mytest'
    },
    {
      'selected': false,
      'expanded': true,
      'loading': false,
      'items': [{
        'selected': true,
        'expanded': false,
        'loading': false,
        'items': [],
        'label': 'My Test',
        'path': '/mytest/test',
        'namespace': 'mytest'
      },
      {
        'selected': false,
        'expanded': false,
        'loading': false,
        'items': [],
        'icon': '',
        'label': 'Other',
        'path': '/other/test',
        'namespace': 'other'
      },
      {
        'selected': false,
        'expanded': false,
        'loading': false,
        'items': [],
        'icon': '',
        'label': 'Other2',
        'path': '/other/test2',
        'namespace': 'other'
      },
      {
        'selected': false,
        'expanded': false,
        'loading': false,
        'items': [],
        'label': 'Other4',
        'path': '/other/test3',
        'namespace': 'other'
      }],
      'icon': IconCode,
      'label': 'Other',
      'path': '/other/test',
      'namespace': 'other'
    },
    {
      'selected': false,
      'expanded': false,
      'loading': false,
      'items': [],
      'icon': IconCluster,
      'label': 'Other2',
      'path': '/other/test2',
      'namespace': 'other'
    },
    {
      'selected': false,
      'expanded': false,
      'loading': false,
      'items': [],
      'icon': IconTrash,
      'label': 'Other4',
      'path': '/other/test3',
      'namespace': 'other'
    },
    {
      'selected': false,
      'expanded': false,
      'loading': false,
      'items': [],
      'icon': IconInfo,
      'label': 'Documentation',
      'path': '/other/test3',
      'namespace': 'doc',
      'type': 'external',
      'pinBottom': true
    }
  ];

  const renderLogo = collapsed => (
    <SVGImage
      className={!collapsed && logoStyle}
      glyph={collapsed ? TarantoolLogoCompact : TarantoolLogoFull}
    />
  );

  const PreconfiguredSideMenu = () => (
    <SideMenu
      menu={menuItems}
      path='/'
      onMenuItemClick={onMenuItemClick}
      toggleExpand={toggleExpand}
      renderMenuLogo={renderLogo}
    />
  );

  const tree = renderer.create(
    <AppLayout sidebarComponent={PreconfiguredSideMenu}>
      <AppHeader
        appName='Sample app'
        breadcrumbs={breadcrumbs}
        controls={[
          <Button key={0} title='Search' intent='plain' icon={IconSearch} />,
          <Button key={1} text='Log in' />
        ]}
        onLinkClick={onLinkClick}
      />
      Content
    </AppLayout>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
