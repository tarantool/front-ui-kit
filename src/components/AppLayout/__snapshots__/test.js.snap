// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLayout renders correctly 1`] = `
<div
  className="css-ww59jg"
>
  <div
    className="css-1d4yvnq"
  >
    <div
      className="css-7s3tqb"
    >
      <svg
        className="css-1d43lrb"
        viewBox="0 0 154 28"
      >
        <use
          xlinkHref="#tarantool-logo-full"
        />
      </svg>
    </div>
    <div
      className="css-htil7a"
    >
      <a
        className="css-1e2guak"
        href="/mytest/test"
        onClick={[Function]}
        title="My Test"
      >
        <div
          className="css-ja26we"
        >
          <svg
            className="css-sffbkr"
            viewBox="0 0 16 16"
          >
            <use
              xlinkHref="#info"
            />
          </svg>
        </div>
        <span
          className="css-lbwipx"
        >
          My Test
        </span>
      </a>
      <a
        className="css-1e2guak"
        href="/other/test"
        onClick={[Function]}
        title="Other"
      >
        <div
          className="css-ja26we"
        >
          <svg
            className="css-1nnk27v"
            viewBox="0 0 14 14"
          >
            <use
              xlinkHref="#code"
            />
          </svg>
        </div>
        <span
          className="css-lbwipx"
        >
          Other
        </span>
        <svg
          className="css-8mr8c5"
          viewBox="0 0 14 14"
        >
          <use
            xlinkHref="#chevron-up"
          />
        </svg>
      </a>
      <div
        className="css-2fcw72"
      >
        <a
          className="css-1ssciey"
          href="/mytest/test"
          onClick={[Function]}
          title="My Test"
        >
          <span
            className="css-bw2ld4"
          >
            My Test
          </span>
        </a>
        <a
          className="css-1e2guak"
          href="/other/test"
          onClick={[Function]}
          title="Other"
        >
          <span
            className="css-oabgn6"
          >
            Other
          </span>
        </a>
        <a
          className="css-1e2guak"
          href="/other/test2"
          onClick={[Function]}
          title="Other2"
        >
          <span
            className="css-oabgn6"
          >
            Other2
          </span>
        </a>
        <a
          className="css-1e2guak"
          href="/other/test3"
          onClick={[Function]}
          title="Other4"
        >
          <span
            className="css-oabgn6"
          >
            Other4
          </span>
        </a>
      </div>
      <a
        className="css-1e2guak"
        href="/other/test2"
        onClick={[Function]}
        title="Other2"
      >
        <div
          className="css-ja26we"
        >
          <svg
            className="css-1e8hhk8"
            viewBox="0 0 14 14"
          >
            <use
              xlinkHref="#cluster"
            />
          </svg>
        </div>
        <span
          className="css-lbwipx"
        >
          Other2
        </span>
      </a>
      <a
        className="css-1e2guak"
        href="/other/test3"
        onClick={[Function]}
        title="Other4"
      >
        <div
          className="css-ja26we"
        >
          <svg
            className="css-1ok0ieg"
            viewBox="0 0 12 14"
          >
            <use
              xlinkHref="#trash"
            />
          </svg>
        </div>
        <span
          className="css-lbwipx"
        >
          Other4
        </span>
      </a>
    </div>
    <div
      className="css-7yk45v"
    >
      <a
        className="css-1e2guak"
        href="/other/test3"
        onClick={[Function]}
        title="Documentation"
      >
        <div
          className="css-ja26we"
        >
          <svg
            className="css-sffbkr"
            viewBox="0 0 16 16"
          >
            <use
              xlinkHref="#info"
            />
          </svg>
        </div>
        <span
          className="css-lbwipx"
        >
          Documentation
        </span>
      </a>
      <button
        className="css-1x7zhyv"
        href=""
        onClick={[Function]}
        title="Collapse menu"
      >
        <div
          className="css-ja26we"
        >
          <svg
            className="css-1l69dq4"
            viewBox="0 0 14 14"
          >
            <use
              xlinkHref="#arrow-left"
            />
          </svg>
        </div>
        <span
          className="css-lbwipx"
        >
          Collapse menu
        </span>
      </button>
    </div>
  </div>
  <div
    className="css-aqdqdm"
  >
    <div
      className="css-1r0gly9"
    >
      <div
        className="css-pavhyf"
      >
        <span
          className="css-10mai1l"
        >
          Sample app
        </span>
        <div
          className="css-1xzbk8"
        >
          <span
            className="css-1pfwcw7"
          >
            /
          </span>
          <a
            className="css-2w80i4"
            href="/tarantool"
            onClick={[Function]}
          >
            @tarantool.io/ui-kit
          </a>
          <span
            className="css-1pfwcw7"
          >
            /
          </span>
          <a
            className="css-2w80i4"
            href="/section-ui-components"
            onClick={[Function]}
          >
            UI Components
          </a>
          <span
            className="css-1pfwcw7"
          >
            /
          </span>
          <a
            className="css-2w80i4"
            href="/breadcrumb"
            onClick={[Function]}
          >
            Breadcrumb
          </a>
          <span
            className="css-1pfwcw7"
          >
            /
          </span>
          <a
            className="css-2w80i4"
            href="/overflow-list"
            onClick={[Function]}
          >
            OverflowList
          </a>
          <span
            className="css-1pfwcw7"
          >
            /
          </span>
          <a
            className="css-2w80i4"
            href="/resize-sensor"
            onClick={[Function]}
          >
            ResizeSensor
          </a>
          <div
            className="css-18izngs"
          />
        </div>
      </div>
      <div
        className="css-ljifeg"
      >
        <div
          className="css-1zje6z"
        >
          <button
            className="css-lot513"
            title="Search"
            type="button"
          >
            <svg
              className="css-10tmi9z"
              viewBox="0 0 16 16"
            >
              <use
                xlinkHref="#search"
              />
            </svg>
          </button>
        </div>
        <div
          className="css-1zje6z"
        >
          <button
            className="css-1mocbyz"
            type="button"
          >
            Log in
          </button>
        </div>
      </div>
    </div>
    Content
  </div>
</div>
`;
